dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs

```diff
@@ -83,20 +84,35 @@ public async Task<Plan> CreatePlanAsync(string goal)
         SKContext result = await this._plannerFunction.InvokeAsync(this._context).ConfigureAwait(false);
 
         ActionPlanResponse? planData;
-        try
+
+        /* Filter out good JSON from the result in case additional text is present:
+        * Follows the balancing group regex defined here: https://learn.microsoft.com/en-us/dotnet/standard/base-types/grouping-constructs-in-regular-expressions#balancing-group-definitions 
+        */
+        Regex planRegex = new("^[^{}]*(((?'Open'{)[^{}]*)+((?'Close-Open'})[^{}]*)+)*(?(Open)(?!))", RegexOptions.Singleline);
+        Match match = planRegex.Match(result.ToString());
+
+        if (match.Success && match.Groups["Close"].Length > 0)
```

Comment:
I would move this logic to separate private method. It will allow to: 
- Replace comments directly in the code with XML documentation above the method, which is responsible for Regex manipulations.
- Reduce code nesting for Regex `if` statement.
- Keep method `CreatePlanAsync` simple.

Reply:
Moved to private `ParsePlannerResult` method